plugins/modules/vmware_object_role_permission_info.py

```diff
@@ -0,0 +1,247 @@
+    ]
+"""
+
+try:
+    from pyVmomi import vim
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.vmware.plugins.module_utils.vmware import (
+    PyVmomi,
+    vmware_argument_spec,
+    find_obj,
+)
+
+
+class VMwareObjectRolePermission(PyVmomi):
+    def __init__(self, module):
+        super(VMwareObjectRolePermission, self).__init__(module)
+        self.module = module
+        self.params = module.params
+        self.role_list = {}
+        self.auth_manager = self.content.authorizationManager
+
+        self.get_object()
+        self.get_perms()
+        self.populate_role_list()
+        self.populate_permission_list()
+
+    def populate_permission_list(self):
+        results = []
+        for permission in self.current_perms:
+            results.append(
+                {
+                    "principal": permission.principal,
+                    "role_name": self.role_list.get(permission.roleId, ""),
+                    "role_id": permission.roleId,
+                    "propagate": permission.propagate,
+                }
+            )
+        self.module.exit_json(changed=False, permission_info=results)
+
+    def populate_role_list(self):
+        user_friendly_role_names = {
+            "Admin": ["Administrator"],
+            "ReadOnly": ["Read-Only"],
+            "com.vmware.Content.Admin": [
+                "Content library administrator (sample)",
+                "Content library administrator",
+            ],
+            "NoCryptoAdmin": ["No cryptography administrator"],
+            "NoAccess": ["No access"],
+            "VirtualMachinePowerUser": [
+                "Virtual machine power user (sample)",
+                "Virtual machine power user",
+            ],
+            "VirtualMachineUser": [
+                "Virtual machine user (sample)",
+                "Virtual machine user",
+            ],
+            "ResourcePoolAdministrator": [
+                "Resource pool administrator (sample)",
+                "Resource pool administrator",
+            ],
+            "VMwareConsolidatedBackupUser": [
+                "VMware Consolidated Backup user (sample)",
+                "VMware Consolidated Backup user",
+            ],
+            "DatastoreConsumer": ["Datastore consumer (sample)", "Datastore consumer"],
+            "NetworkConsumer": [
+                "Network administrator (sample)",
+                "Network administrator",
+            ],
+            "VirtualMachineConsoleUser": ["Virtual Machine console user"],
+            "InventoryService.Tagging.TaggingAdmin": ["Tagging Admin"],
+        }
+        for role in self.content.authorizationManager.roleList:
+            self.role_list[role.roleId] = role.name
+            if user_friendly_role_names.get(role.name):
+                for role_name in user_friendly_role_names[role.name]:
+                    self.role_list[role.roleId] = role.name
```

Comment:
```suggestion
                for role_name in user_friendly_role_names[role.name]:
                    self.role_list[role.roleId] = role_name
```
This should be role_name not role.name, shouldn’t it?